File: 2026/05-May/17-Sunday/solution.go
func canReach(arr []int, start int) bool {

    if start < 0 || start >= len(arr) || arr[start] < 0 {
        return false
    }

    if arr[start] == 0 {
        return true
    }

    jump := arr[start]
    arr[start] = -arr[start]

    return canReach(arr, start+jump) ||
           canReach(arr, start-jump)
}

File: 2026/05-May/17-Sunday/solution.cpp
class Solution {
public:
    bool dfs(vector<int>& arr, int idx) {
        if (idx < 0 || idx >= arr.size() || arr[idx] < 0)
            return false;

        if (arr[idx] == 0)
            return true;

        int jump = arr[idx];
        arr[idx] = -arr[idx];   // mark visited

        return dfs(arr, idx + jump) ||
               dfs(arr, idx - jump);
    }

    bool canReach(vector<int>& arr, int start) {
        return dfs(arr, start);
    }
};
